cdap-app-fabric/src/main/java/co/cask/cdap/app/mapreduce/MRJobClient.java

```diff
@@ -82,6 +82,10 @@ public MRJobInfo getMRJobInfo(Id.Run runId) throws IOException, NotFoundExceptio
     JobStatus thisJob = findJobForRunId(jobs, runId);
 
     RunningJob runningJob = jobClient.getJob(thisJob.getJobID());
+    if (runningJob == null) {
+      LOG.warn("JobClient returned null for JobId: '{}'", thisJob.getJobID());
+      throw new NotFoundException(runId);
```

Comment:
I think the NFE is also appropriate, since the JHS couldn't retrieve the job for the particular runId, but IllegalStateException or IOException also seem to fit.